### PR TITLE
Pract.02: Add 07 | Ivan Stoynev

### PR DESCRIPTION
**Pull request by Vacata12**
Ivan Stoynev
IS 3 6MI0700164

### CODE REVIEW

File: Seminars/Sem.02/Pract.02/Task07.cpp [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    while(arr[count] != '\0')
+    {
+        count += 1;
+    }
+    return count;
+}
+
+void loadUsersFromFile(System& s)
+{
+    std::fstream file(SYSTEM_FILE_NAME);
```
Comment (Justsvetoslavov) [inline Seminars/Sem.02/Pract.02/Task07.cpp]: why fstream?

File: Seminars/Sem.02/Pract.02/Task07.cpp [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    }
+    return false;
+}
+
+bool myStrCompare(const char* arr1, const char* arr2)
+{
+    if(myStrLen(arr1) != myStrLen(arr2))
+        return false;
+    int index = 0;
+    while(arr1[index] != '\0' || arr2[index] != '\0')
```
Comment (Justsvetoslavov) [inline Seminars/Sem.02/Pract.02/Task07.cpp]: if strlen is same , why double check?

File: Seminars/Sem.02/Pract.02/Task07.cpp [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    while(name[index] != '\0')
+    {
+        if(!('a' <= name[index] && name[index] <= 'z') && !('A' <= name[index] && name[index] <= 'Z') && name[index] != ' ')
+            return false;
+        index += 1;
+    }
+    return true;
+}
+
+bool validEmail(const char* email)
```
Comment (Justsvetoslavov) [inline Seminars/Sem.02/Pract.02/Task07.cpp:109]: If we have a validation email, only count is not enough as logic.

File: Seminars/Sem.02/Pract.02/Task07.cpp [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+};
+
+void registerUser(User& user);
+bool validName(const char* name);
+bool validEmail(const char* email);
+void writeUserToFile(const User& user);
+void readUserFromFile();
+bool userInSystem(System s, const unsigned short registeredUsers);
+bool myStrCompare(const char* arr1, const char* arr2);
+unsigned int myStrLen(const char*& arr);
```
Comment (Justsvetoslavov) [inline Seminars/Sem.02/Pract.02/Task07.cpp]: reference to a pointer -> why ?

File: Seminars/Sem.02/Pract.02/Task07.cpp [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+    std::cout << "Enter email: " << '\n'; 
+    do
+    {
+         std::cin.getline(user.email, MAX_SIZE + 1);
+        if(!validEmail(user.email))
+            std::cout << "Invalid email \n";
+    } while (!validEmail(user.email));
+    
+    std::cout << "Enter password: " << '\n'; 
+    std::cin.getline(user.password, MAX_SIZE + 1);
```
Comment (Justsvetoslavov) [inline Seminars/Sem.02/Pract.02/Task07.cpp]: validation of password?

File: Seminars/Sem.02/Pract.02/Task07.cpp [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+        if(email[index] == '@')
+            count += 1;
+        index += 1;
+    }
+    return count != 0;
+}
+
+void writeUserToFile(const User& user)
+{
+    std::fstream file(SYSTEM_FILE_NAME);
```
Comment (Justsvetoslavov) [inline Seminars/Sem.02/Pract.02/Task07.cpp]: why fstream?

File: Seminars/Sem.02/Pract.02/Task07.cpp [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+int main()
+{
+    System s;
+    loadUsersFromFile(s);
+    char command[COMMAND_SIZE + 1];
+    do
+    {
+        std::cout << "Enter command \n";
+        std::cin.getline(command, COMMAND_SIZE + 1);
+        if(myStrCompare(command, "register"))
```
Comment (Justsvetoslavov) [inline Seminars/Sem.02/Pract.02/Task07.cpp]: Can we use a switch case? Or if-else is fine?

**Review by Justsvetoslavov — CHANGES_REQUESTED** [review]

File: Seminars/Sem.02/Pract.02/Task07.cpp (hunk 1 above)
Comment (Vacata12) [reply Seminars/Sem.02/Pract.02/Task07.cpp:191]: I should open it with ofstream, but I have read that it works that way if the file already existed it deletes the file and the data in it and creates a new one.

File: Seminars/Sem.02/Pract.02/Task07.cpp (hunk 6 above)
Comment (Vacata12) [reply Seminars/Sem.02/Pract.02/Task07.cpp]: I should open it with ofstream, but I have read that it works that way if the file already existed it deletes the file and the data in it and creates a new one with the same name.

Comment (Justsvetoslavov) [reply Seminars/Sem.02/Pract.02/Task07.cpp]: Check the available flags, you are missing the create/nocreate flags.

**Pull request merged**